tasks/definitions/container/start.js

```diff
@@ -0,0 +1,141 @@
+const { spawnCmd } = require('@keg-hub/spawn-cmd')
+const { getConfig } = require('../../utils/getConfig')
+const { values } = require('../constants/values')
+const { addParam, addFlag, addValues } = require('../../utils/cmd')
+const { attach: attachTask } = require('./attach')
+const os = require('os')
+const { exists } = require('@keg-hub/jsutils/build/cjs')
+
+// task parameter defaults
+const paramDefaults = {
+  debug: false,
+  runOnce: false,
+  debug: false,
+  cleanup: false,
+  interval: 5*60,
+  rm: true
+}
+
+/**
+ * Builds the docker run command for watchtower, given the parameters
+ * @param {Object} params 
+ * @returns {string} the command
+ */
+const buildCmd = params => {
+  const dockerSocketPath = '/var/run/docker.sock'
+  const dockerRegistryConfigPath = `${os.homedir()}/.docker/config.json`
+
+  const {
+    dockerApiSock=dockerSocketPath,
+    configPath=dockerRegistryConfigPath,
+    cleanup,
+    debug,
+    runOnce,
+    containers,
+    rm,
+    interval=2*60 // every 2 minutes
+  } = params
+
+  return `docker run \
+    ${addFlag('rm', rm)} \
+    -d \
+    -it \
+    --name ${values.containerName} \
+    -v ${dockerApiSock}:${dockerSocketPath} \
+    -v ${configPath}:/config.json \
+    containrrr/watchtower \
+    ${addFlag('cleanup', cleanup)} \
+    ${addFlag('debug', debug)} \
+    ${addFlag('run-once', runOnce)} \
+    ${addParam('interval', interval)} \
+    ${addValues(containers)}`
+}
+
+/**
+ * @param {Object} taskParams - parameters defined at the task definition
+ * @returns {Object} the parameters to use in the task function, derived from config files and command line
+ */
+const getResolvedParams = taskParams => {
+  const { 
+    config: configPath, 
+    ...cmdParams 
+  } = taskParams
+
+  const config = getConfig(configPath)
+
+  // Merge configs together, with precedence order: 
+  //  command line task parameters, followed by config files, followed by defaults
+  const sources = [ cmdParams, config, paramDefaults ]
+  return Object.keys(cmdParams).reduce(
+    (acc, key) => {
+      // find the first source that has a **defined** key 
+      const source = sources.find(src => exists(src[key])) || {}
+      acc[key] = source[key]
+      return acc
+    },
+    {}
+  )
+}
+
+/**
+ * Run watchtower container, configured by task parameters and global
+ * watchtower.config.js(on) files
+ * @param {Object} args 
+ */
+const startWatchtower = async args => {
+  const params = getResolvedParams(args.params)
+
+  const cmd = buildCmd(params)
+  params.debug && console.log(cmd.replace(/\s+/g, ' '))
+
+  await spawnCmd(cmd)
+
+  params.attach && attachTask.action()
+}
+
+
+
+module.exports = {
+  start: {
+    name: 'start',
+    action: startWatchtower,
+    example: 'keg watchtower start',
+    description : 'Starts the watchtower container',
+    alias: ['setup', 'init'],
+    options: {
```

Comment:
We should add a `--restart` option, with the default being `unless-stopped`
  * See [docker docs here](https://docs.docker.com/config/containers/start-containers-automatically/)
  * This way if it fails on the staging server, it will restart it self automatically